druid/src/core.rs

```diff
@@ -559,160 +559,173 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                 // but was hot previously. This is to allow the widget to respond to the movement,
                 // e.g. drag functionality where the widget wants to follow the mouse.
-                recurse = had_active || child_ctx.widget_state.is_hot || hot_changed;
-                let mut mouse_event = mouse_event.clone();
-                mouse_event.pos -= rect.origin().to_vec2();
-                Event::MouseMove(mouse_event)
+                if had_active || self.state.is_hot || hot_changed {
+                    let mut mouse_event = mouse_event.clone();
+                    mouse_event.pos -= rect.origin().to_vec2();
+                    modified_event = Some(Event::MouseMove(mouse_event));
+                    true
+                } else {
+                    false
+                }
             }
             Event::Wheel(mouse_event) => {
-                WidgetPod::set_hot_state(
+                needs_merge |= WidgetPod::set_hot_state(
                     &mut self.inner,
-                    child_ctx.widget_state,
-                    child_ctx.state,
+                    &mut self.state,
+                    ctx.state,
                     rect,
                     Some(mouse_event.pos),
                     data,
                     env,
                 );
-                recurse = had_active || child_ctx.widget_state.is_hot;
-                let mut mouse_event = mouse_event.clone();
-                mouse_event.pos -= rect.origin().to_vec2();
-                Event::Wheel(mouse_event)
-            }
-            Event::KeyDown(e) => {
-                recurse = child_ctx.has_focus();
-                Event::KeyDown(*e)
-            }
-            Event::KeyUp(e) => {
-                recurse = child_ctx.has_focus();
-                Event::KeyUp(*e)
-            }
-            Event::Paste(e) => {
-                recurse = child_ctx.has_focus();
-                Event::Paste(e.clone())
-            }
-            Event::Zoom(zoom) => {
-                recurse = had_active || child_ctx.widget_state.is_hot;
-                Event::Zoom(*zoom)
-            }
-            Event::Timer(token) => {
-                recurse = false;
-                Event::Timer(*token)
+                if had_active || self.state.is_hot {
+                    let mut mouse_event = mouse_event.clone();
+                    mouse_event.pos -= rect.origin().to_vec2();
+                    modified_event = Some(Event::Wheel(mouse_event));
+                    true
+                } else {
+                    false
+                }
             }
-            Event::Command(cmd) => Event::Command(cmd.clone()),
+            Event::KeyDown(_) => self.state.has_focus,
+            Event::KeyUp(_) => self.state.has_focus,
+            Event::Paste(_) => self.state.has_focus,
+            Event::Zoom(_) => had_active || self.state.is_hot,
+            Event::Timer(_) => false, // This event was targeted only to our parent
```

Comment:
I think this filtering of events passed to children should be documented on the `WidgetPot::event` method.